File: common/src/main/kotlin/net/impleri/playerskills/skills/registry/storage/PersistentStorage.kt
package net.impleri.playerskills.skills.registry.storage

import java.io.File

/**
 * Manages _how_ to save data
 */
internal interface PersistentStorage {
  fun read(file: File): List<String>
  fun write(file: File, skills: List<String>)
}

File: common/src/main/scala/net/impleri/playerskills/skills/registry/storage/PersistentStorage.scala
package net.impleri.playerskills.skills.registry.storage

import java.io.File

trait PersistentStorage {
  def read(file: File): Either[NbtFileReadError, List[String]]

  def write(file: File, skills: List[String]): Either[NbtFileWriteError, Boolean]
}
